File: core/documents/intelligence/code_block_engine.py
from __future__ import annotations
import re

def extract_code_blocks(text: str):
    blocks=[b.strip() for b in re.findall(r"```[\w-]*\n(.*?)```", text or '', flags=re.DOTALL) if b.strip()]
    return {"code_blocks": sorted(blocks)}

File: src/documents/intelligence/codeBlockEngine.ts
/**
 * Converted from Python: core/documents/intelligence/code_block_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";

export function extractCodeBlocks(text: any): any {
  var blocks: any = py.iter(py.reFindall("```[\\w-]*\\n(.*?)```", py.or2(text, () => ("")), "s")).filter((b: any) => py.truthy(py.strip(b))).map((b: any) => py.strip(b));
  return {"code_blocks": py.sorted(blocks)};
}
